lib/request_started_on_middleware.rb

```diff
@@ -0,0 +1,59 @@
+# Add to config/application.rb:
+#
+#     config.middleware.use 'RequestStartedOnMiddleware'
+#
+class RequestStartedOnMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    begin
+      start_request(env['PATH_INFO'], Time.now.utc)
+      response = @app.call(env)
+    ensure
+      complete_request
+    end
+
+    response
+  end
+
+  def start_request(path, started_on)
+    Thread.current[:current_request] = path
+    Thread.current[:current_request_started_on] = started_on
+  end
+
+  def complete_request
+    Thread.current[:current_request] = nil
+    Thread.current[:current_request_started_on] = nil
+  end
+
+  def self.long_running_requests
+    requests = []
+    relevant_thread_list.each do |thread|
+      request    = thread[:current_request]
+      started_on = thread[:current_request_started_on]
+
+      # There's a race condition where the complete_request method runs in another
+      # thread after we set one or more of the above local variables. The fallout
+      # of this is we return a false positive for a request that finished very close
+      # to the 2 minute timeout.
+      if request.present? && started_on.kind_of?(Time) && REQUEST_TIMEOUT.ago > started_on
```

Comment:
Was this supposed to use the class method rather than the constant?

Reply:
👀 YES!  Thanks!

Reply:
Refactoring for testing and forgetting to call the new method... It's a constant problem.